### ISSUE TITLE
Hide [x-cloak] only when JS is available

### ISSUE THREAD

**Issue by klickreflex** [studio1902/statamic-peak#117]
**Is your feature request related to a problem? Please describe.**
Setting `[x-cloak]` to hidden makes Alpine components inaccessible when JS fails  or is disabled. As some of these components might still work without JS I suggest only hiding it when JS is available.

(Related: we could also add a no-js fallback to the dropown navigation by using the parent selector.)

**Describe the solution you'd like**
1. Add a class of `no-js` to the root element
2. Add a script to replace that with `js` (which obviously won't happen without JS): 
`document.documentElement.className = document.documentElement.className.replace('no-js', 'js');`
3. Prepend a parent class of `.js`  to the `[x-cloak]` selector used to hide the component: (`.js [x-cloak]`) 

**Describe alternatives you've considered**
– 

**Additional context**
–

**Comment by robdekort** [comment]
To be completely honest I'm not sure if I want to take it this far since it's a bit of a slippery slope. Few sites work without JS and most stuff on my sites will break when JS is off anyway. Navigation, mobile navigation, form sending and all custom alpine modules. 

Another worry I have is that this might cause the flash that x-cloak prevents to reappear. 

However I do think it's important to make stuff accessible for as many folks as possible. 

I'm curious how other users feel about this. Maybe someone wants to pitch in.

**Comment by sjclark** [comment]
> I'm curious how other users feel about this. Maybe someone wants to pitch in.

I think making everything run without JS is an extremely noble and great thing to do! **BUT** it's also catering very much to the very distinct minority, usually results in hacks that might actually degrade the experience for the average JS-using user. Virtually any site I've made in the past few years (peak or peak-less) would be significantly handicapped without JS.

Nothing inherently wrong with the request above, but it's just a bit of a band-aid to a specific issue, and is in isolation to all the other parts of Peak that use JS.

The boilerplate code can't accommodate everyone - alternative might just be to add a tiny bit to the documentation to the peak docs outlining how the JS works from a top-level, and recommendations of the things you'd have to do (or limitations) if you don't want to use JS in your specific situation/usage?

**Comment by klickreflex** [comment]
I agree that we can't expect _everything_ to work without JS. However, I personally try to make at least every page accessible, that is, having at least some form of navigation present, even if it's just a long list of links that's better than nothing.

The other thing when taking about "no-js" use cases I want to mention is that I'm mostly not concerned about that tiny percentage of users who access the web without JS on purpose. 

I'm more concerned to prepare for situations where JS fails, be it because of an error somewhere else in your bundle or because a network request times out. Happens more often than one might think.

This is a great flowchart regarding the question: [Everyone has JavaScript, right?](https://kryogenix.org/code/browser/everyonehasjs.html)

> Another worry I have is that this might cause the flash that x-cloak prevents to reappear.

That's a very good point, I'll keep an eye on it.

**Comment by robdekort** [comment]
Thanks for your thoughts both! Let me know what you find re: the flash It's usually very obvious.

**Comment by klickreflex** [comment]
> Thanks for your thoughts both! Let me know what you find re: the flash It's usually very obvious.

You're right: I do get a short flash of the supposedly hidden element. It's shorter than waiting for Alpine's initialization, but it still sucks and defeats the purpose of x-cloak. I guess I need to find a better approach for no-js fallbacks.

**Comment by robdekort** [comment]
Ah yeah, then it does indeed defeat the purpose. Thanks folks!

**Comment by robdekort** [comment]
Thinking of this I agree that we should try to enforce progressive enhancement better. How about this solution:

```html
<noscript>
    <style>
        [x-cloak] {
            display: block !important;
        }
    </style>
</noscript>
```

**Comment by robdekort** [comment]
Could be as simple as this in the `<head>`:

`{{ partial:snippets/noscript }}`

I really like this simple fix! Am I missing something here?

**Comment by robdekort** [comment]
Closed in #120

**Comment by robdekort** [comment]
Took me way too long as well. Glad you like it.
